src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -355,6 +355,15 @@ private static void CleanupSourceMappingStart(FormattingContext context, Range s
             return;
         }
 
+        // Parent.Parent.Parent is because the tree is
+        //  ExplicitExpression -> ExplicitExpressionBody -> CSharpCodeBlock -> CSharpExpressionLiteral
+        if (owner is CSharpExpressionLiteralSyntax { Parent.Parent.Parent: CSharpExplicitExpressionSyntax explicitExpression } &&
```

Comment:
This change is required because without code actions, there was no real way to invoke ontype formatting in an explicit expression, so there was just no code to deal with them.